app/controllers/answers_controller.rb

```diff
@@ -1,24 +1,38 @@
 class AnswersController < ApplicationController
+  before_action :authenticate_user!, except: :show
   before_action :load_question
+  before_action :load_answer, only: :destroy
 
   def new
     @answer = @question.answers.new
   end
 
   def create
-    @answer = @question.answers.create(answer_params)
+    @answer = @question.answers.create answer_params.merge(user: current_user)
 
     if @answer.save
-      redirect_to question_answers_path(@question)
+      redirect_to question_path(@question)
+      flash[:notice] = 'Your answer submitted'
     else
       render :new
+      flash[:warning] = 'Oops! Your answer missed'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.